design-toolkit/components/src/components/tree/index.tsx

```diff
@@ -0,0 +1,365 @@
+    const type = i === level - 1 ? 'branch' : 'vert';
+    const line = isLastOfSet && i > 0 ? 'last' : type;
+    const size = variant === 'cozy' ? 'large' : 'medium';
+
+    return (
+      <Lines
+        key={i}
+        variant={line}
+        size={size}
+        showLines={showRuleLines}
+        className={spacing({ variant })}
+      />
+    );
+  });
+});
+
+function reducer<T, I>(
+  nodes: TreeNode<T>[],
+  callback: (acc: I, node: TreeNode<T>) => I,
+  initial: I,
+): I {
+  return nodes.reduce(
+    (acc, node) => reducer(node.children ?? [], callback, callback(acc, node)),
+    initial,
+  );
+}
+
+export function Tree<T>(props: TreeProps<T>) {
+  const {
+    children,
+    className,
+    variant = TreeStylesDefaults.variant,
+    selectionMode = 'multiple',
+    visibleKeys,
+    onVisibilityChange,
+    showRuleLines = true,
+    showVisibility = true,
+    dragAndDropConfig,
+    items,
+    ...rest
+  } = props;
+
+  const { dragAndDropHooks } = useDragAndDrop({
+    getItems: dragAndDropConfig?.getItems,
+    onReorder: dragAndDropConfig?.onReorder,
+    onRootDrop: dragAndDropConfig?.onRootDrop,
+    renderDragPreview: dragAndDropConfig?.renderDragPreview,
+    renderDropIndicator:
+      dragAndDropConfig?.renderDropIndicator ?? defaultRenderDropIndicator,
+    acceptedDragTypes: dragAndDropConfig?.acceptedDragTypes,
+    getAllowedDropOperations: () => ['move'],
+    getDropOperation: () => 'move',
+    onInsert: dragAndDropConfig?.onInsert,
+    onItemDrop: dragAndDropConfig?.onItemDrop,
+  });
+
+  const viewableKeys = useMemo(
+    () =>
+      reducer(
+        Array.from(items ?? []),
+        (acc, node) => {
+          return node.isViewable ? acc.add(node.key) : acc;
+        },
+        new Set<Key>(),
+      ),
+    [items],
+  );
+
+  return (
+    <TreeContext.Provider
+      value={{
+        showRuleLines,
+        showVisibility,
+        visibleKeys,
+        viewableKeys,
+        onVisibilityChange,
+        variant,
+        isStatic: !items,
+      }}
+    >
+      <AriaTree
+        selectionMode={selectionMode}
+        dragAndDropHooks={dragAndDropHooks}
+        className={composeRenderProps(className, (className) =>
+          tree({ className, variant }),
+        )}
+        items={items}
+        {...rest}
+      >
+        {children}
+      </AriaTree>
+    </TreeContext.Provider>
+  );
+}
+Tree.displayName = 'Tree';
+
+export function TreeItem(props: TreeItemProps) {
+  const { id, children, className, label, isDisabled, ...rest } = props;
+
+  return (
+    <AriaTreeItem
+      id={id}
+      className={composeRenderProps(className, (className) =>
+        item({ className, isDisabled }),
+      )}
+      textValue={label}
+      {...rest}
+    >
+      {children}
+    </AriaTreeItem>
+  );
+}
+TreeItem.displayName = 'Tree.Item';
+
+export function ItemContent({ children }: ItemContentProps) {
+  const context = useContext(TreeContext);
```

Comment:
Since this context is never used as a props context, it doesn't need to conform to the RAC context type.

The only reason I created this pattern in another component was when the component both consumed AND provided the same context. If you're not doing that then this pattern just makes life harder